src/shared/services/HttpService.ts

```diff
@@ -0,0 +1,92 @@
+import { LemmyHttp } from "lemmy-js-client";
+import { getHttpBase } from "../../shared/env";
+import { i18n } from "../../shared/i18next";
+import { toast } from "../../shared/utils";
+
+type EmptyRequestState = {
+  state: "empty";
+};
+
+type LoadingRequestState = {
+  state: "loading";
+};
+
+type FailedRequestState = {
+  state: "failed";
+  msg: string;
+};
+
+type SuccessRequestState<T> = {
+  state: "success";
+  data: T;
+};
+
+/**
+ * Shows the state of an API request.
+ *
+ * Can be empty, loading, failed, or success
+ */
+export type RequestState<T> =
+  | EmptyRequestState
+  | LoadingRequestState
+  | FailedRequestState
+  | SuccessRequestState<T>;
+
+export async function apiWrapper<ResponseType>(
```

Comment:
It's kind of annoying needing to use a wrapper around pretty much every call with the client. I think it might be good to move this logic into the js client project.

Reply:
I'm still fairly torn about it. Yes its annoying, but this seems almost like application-specific logic for how to handle API state: errors, loading indicators, etc. 

It might be a bit strange to add this directly into the current API wrapper in lemmy-js-client, and have that enum variant / discriminated union be required by anyone who uses lemmy-js-client. They might have their own method of handling API state.

Reply:
This comment change was resolved by centralizing the API wrapping logic in `HttpService.ts`.